File: nomad-samples/python/basic-live-channel-1.1/live_input/delete_live_input.py
from constants.project_constants import *
from exceptions.api_exception_handler import *
from live_input.live_input_statuses import *
from live_input.wait_live_input_status import *

from libraries import json, requests

'''
 * Delete Live Input
 *
 * NOTE: Live inputs being used in live channels can't be deleted
 *
 * @param {string} AUTH_TOKEN    | Authentication token
 * @param {string} INPUT_ID      | The ID of the input to delete
'''
async def delete_live_input(AUTH_TOKEN, INPUT_ID):
    # Check for valid parameters
    if (not AUTH_TOKEN or not INPUT_ID):
        raise Exception("Delete Live Input: Invalid API call")


    # Create header for the request
    HEADERS = {
        "Authorization": "Bearer " + AUTH_TOKEN
    }

    try:
        # Send the request
        RESPONSE = requests.delete(SERVER_URL + "/liveInput/" + INPUT_ID, headers= HEADERS)

        # Wait for the live input to be deleted
        await wait_for_live_input_status(AUTH_TOKEN, INPUT_ID, LIVE_INPUT_STATUSES["Deleted"], 60, 2)

        # Return the JSON response
        return json.loads(RESPONSE.text)

    except:

        await api_exception_handler(RESPONSE, "Delete Live Input " + INPUT_ID + " failed")

File: nomad-samples/admin/js/basic-live-channel/live-input/delete-live-input.js
import * as prjConstants from "../constants/project-constants.js";
import apiExceptionHandler from "../exceptions/api-exception-handler.js";

/**
 * Delete Live Input
 *
 * NOTE: Live inputs being used in live channels can't be deleted
 *
 * @param {string} authToken    | Authorization token
 * @param {string} inputId      | The ID of the input to delete
 */
export default async function deleteLiveInput(authToken, inputId) {
    // Create header for the request
    const HEADERS = new Headers();
    HEADERS.append("Authorization", `Bearer ${authToken}`);

    // Send the request
    const RESPONSE = await fetch(`${prjConstants.ADMIN_API_URL}/liveInput/${inputId}`, {
        method: "DELETE",
        headers: HEADERS
    });

    // Check for success
    if (RESPONSE && RESPONSE.ok) {

        // Return the JSON response
        return;
    }

    await apiExceptionHandler(RESPONSE, `Delete Live Input ${inputId} failed`);
}
